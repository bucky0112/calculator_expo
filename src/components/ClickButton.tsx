import {
  Text,
  TouchableOpacity,
  TextProps,
  TouchableOpacityProps,
} from 'react-native'
import classNames from 'classnames'

type ClickButtonProps = TouchableOpacityProps & {
  text: string
  size?: 'double'
  theme?: 'primary' | 'secondary' | 'accent'
  textProps?: TextProps
}

const ClickButton = ({
  onPress,
  text,
  size,
  theme = 'primary',
  textProps,
  ...rest
}: ClickButtonProps) => {
  

  return (
    <TouchableOpacity
      onPress={onPress}
      className={classNames({
        "rounded-full flex-1 items-center justify-center m-2": true,
        "h-[84px]": true,
        "w-[84px]": true,
        'w-full': size === 'double',
        'bg-primary-btn': theme === 'primary',
        'bg-secondary-btn': theme === 'secondary',
        'bg-accent-btn': theme === 'accent'
      })}
      {...rest}
      testID='click-button'
    >
      <Text
        className={classNames({
          'text-3xl font-bold text-center': true,
          'text-white': theme === 'secondary' || theme === 'primary',
        })}
        {...textProps}
      >
        {text}
      </Text>
    </TouchableOpacity>
  )
}

export { ClickButton }
