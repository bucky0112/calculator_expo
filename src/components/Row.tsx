import { View } from 'react-native'

type RowProps = {
  children: React.ReactNode;
};

const Row = ({ children, ...rest }: RowProps) => {
  return (
    <View className='flex-row items-center gap-1' {...rest}>
      {children}
    </View>
  )
}

export { Row }
