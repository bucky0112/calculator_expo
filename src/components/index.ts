export { ClickButton } from './ClickButton'
export { Row } from './Row'
