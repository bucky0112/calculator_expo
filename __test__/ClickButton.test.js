import { ClickButton } from "../src/components"
import { render, fireEvent } from "@testing-library/react-native"

describe("ClickButton", () => {
  it("should render correctly", () => {
    const { toJSON } = render(<ClickButton />)
    expect(toJSON()).toMatchSnapshot()
  })

  it("should call onPress when clicked", () => {
    const onPress = jest.fn()
    const { getByTestId } = render(<ClickButton onPress={onPress} />)
    fireEvent.press(getByTestId("click-button"))
    expect(onPress).toHaveBeenCalled()
  })
})
