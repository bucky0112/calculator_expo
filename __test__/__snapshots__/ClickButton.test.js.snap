// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClickButton should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#333333",
      "borderBottomLeftRadius": 9999,
      "borderBottomRightRadius": 9999,
      "borderTopLeftRadius": 9999,
      "borderTopRightRadius": 9999,
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "height": 84,
      "justifyContent": "center",
      "marginBottom": 8,
      "marginLeft": 8,
      "marginRight": 8,
      "marginTop": 8,
      "opacity": 1,
      "width": 84,
    }
  }
  testID="click-button"
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 30,
          "lineHeight": 36,
        },
        Object {
          "fontWeight": "700",
        },
        Object {
          "textAlign": "center",
        },
        Object {
          "color": "#fff",
        },
      ]
    }
  />
</View>
`;
